cmd/snap/cmd_run.go

```diff
@@ -954,3 +963,68 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		return syscallExec(cmd[0], cmd, env)
 	}
 }
+
+func randomUUID() (string, error) {
+	uuidBytes, err := ioutil.ReadFile("/proc/sys/kernel/random/uuid")
+	return strings.TrimSpace(string(uuidBytes)), err
+}
+
+var createTransientScope = func(securityTag string) error {
+	if !features.RefreshAppAwareness.IsEnabled() {
+		return nil
+	}
+
+	// The scope is created with a DBus call to systemd running either on
+	// system or session bus, depending on if we are starting a program as root
+	// or as a regular user.
+	var conn *dbus.Conn
+	var err error
+	if os.Getuid() == 0 {
+		conn, err = dbus.SystemBus()
+	} else {
+		conn, err = dbus.SessionBus()
+	}
+	if err != nil {
+		return err
+	}
+
+	type property struct {
+		Name  string
+		Value interface{}
+	}
+	type auxUnit struct {
+		Name  string
+		Props []property
+	}
+
+	// We ask the kernel for a random UUID. We need one because each transient
+	// scope needs a unique name. The unique name is comprosed of said UUID and
+	// the snap security tag.
+	uuid, err := randomUUID()
```

Comment:
We discussed this idea in during interactive review and I expressed my preference for counters but there are some reservations:

- counters would look much nicer
- counters would have to be stored in ~HOME/ - where exactly?
- perhaps ~/snap/$SNAP_INSTANCE_NAME/.run-counter?
  - privacy concerns?
- perhaps $XGD_RUNTIME_DIR/snapd/$SNAP_INSTANCE_NAME.counter
- some sessions don't have $XDG_RUNTIME_DIR

There would have to be a locking protocol (simple flock would be sufficient) to read and update the counter file.

The UUID approach has some downsides as well:
- it is arguably less nice
- it may require entropy (to be verified by reading kernel source)

Reply:
I added a TODO comment to keep track of this.